### ISSUE TITLE
[Bug]: `ts_project` validation fails on Windows for paths with special characters

### ISSUE THREAD

**Issue by willjschmitt** [aspect-build/rules_ts#666]
### What happened?

When building a library ts_project within a Next.js app that has special characters (e.g. for route grouping), the ts_project validation fails with a bash syntax error from the unescaped special characters in the file path (in my case `(marketing)`

```
ERROR: C:/users/willschmitt/documents/achillea/achillea/www/achillearesearch_com/app/(marketing)/BUILD.bazel:3:11: TsValidateOptions www/achillearesearch_com/app/(marketing)/(marketing).optionsvalid.d.ts failed: (Exit 2): validator.bat failed: error executing TsValidateOptions command (from target //www/achillearesearch_com/app/(marketing):(marketing))
  cd /d C:/_bzl/execroot/_main
  SET BAZEL_BINDIR=bazel-out/x64_windows-fastbuild/bin
  bazel-out\x64_windows-opt-exec-ST-13d3ddad9198\bin\external\aspect_rules_ts~~ext~npm_typescript\validator.bat www/achillearesearch_com/app/(marketing)/tsconfig_(marketing).json www/achillearesearch_com/app/(marketing)/(marketing).optionsvalid.d.ts @@//www/achillearesearch_com/app/(marketing):(marketing) www/achillearesearch_com/app/(marketing) {"allow_js":false,"composite":true,"declaration":true,"declaration_map":false,"emit_declaration_only":false,"incremental":true,"preserve_jsx":false,"resolve_json_module":true,"source_map":false,"ts_build_info_file":""}
# Configuration: 21d0e975c9b4032915c05f5c2f059dd025b6cc6feb8901ed4ba494dc16c40af5
# Execution platform: @@local_config_platform//:host
'DOSKEY' is not recognized as an internal or external command,
operable program or batch file.
/usr/bin/bash: -c: line 1: syntax error near unexpected token `('
/usr/bin/bash: -c: line 1: `C:/_bzl/execroot/_main/bazel-out/x64_windows-opt-exec-ST-13d3ddad9198/bin/external/aspect_rules_ts~~ext~npm_typescript/validator.sh www/achillearesearch_com/app/(marketing)/tsconfig_(marketing).json www/achillearesearch_com/app/(marketing)/(marketing).optionsvalid.d.ts @@//www/achillearesearch_com/app/(marketing):(marketing) www/achillearesearch_com/app/(marketing) "{\\"allow_js\\":false,\\"composite\\":true,\\"declaration\\":true,\\"declaration_map\\":false,\\"emit_declaration_only\\":false,\\"incremental\\":true,\\"preserve_jsx\\":false,\\"resolve_json_module\\":true,\\"source_map\\":false,\\"ts_build_info_file\\":\\"\\"}"'
```

### Version

Development (host) and target OS/architectures: Windows 10 64bit

Output of `bazel --version`: 7.1.1

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
 * aspect_rules_ts - 2.4.2
 * aspect_bazel_lib - 2.7.9 (upgraded after having other path escaping issues from file copying in earlier v1.x releases)

Language(s) and/or frameworks involved: Typescript and Next.js

### How to reproduce

```shell
Build a `ts_project` within a Bazel package with parentheses in the path on Windows with `validate` set to `True`
```

### Any other information?

For now, I'm disabling validation on ts_project targets under directories with special characters on Windows with a select statement, and getting coverage of validation via Linux CI, but it would be better to have immediate feedback during development:
```starlark
ts_project(
    ...
    validate = select({
        "@platforms/os:windows": False,
        "//conditions:default": True,
    }),
)
```

I also started with a similar issue of unrecognized programs when I was on v1 of the Aspect bazel-lib, but the rules since moved from .bat execution to coreutils execution of CopyFile. I wonder if there is a generalized issue for path normalization on Windows for the bazel-lib

**Comment by willjschmitt** [comment]
Pushing a little further, this might be a general issue, with also the transpilation step

**Comment by willjschmitt** [comment]
Both of the executed targets that are failing are `js_binary` targets, so this might end up being a `rules_js` bug

**Comment by jbedard** [comment]
Is this still an issue?

**Comment by willjschmitt** [comment]
Yep. this still appears to be an issue with rules_ts == 3.5.7 and rules_js == 2.3.7

I might take a stab to go find the root cause of this bug and see if it's easily patchable. I'm going to guess it's something that can be handled with the right quoting

**Comment by willjschmitt** [comment]
Digging in a little bit more, it looks like this comes from that bazel single-quotes arguments on linux that it passes directly to the bash script, while on Windows the arguments are unquoted. Those unquoted args are then passed into `bash -c ` in `validator.bat`, which fails on the call to bash trying to expand the parens. If I quote just the paths alone, the validator js script still failed to parse the json input as an argument since the curly braces were parsed by bash as well, but that might be from how I'm invoking it from windows command prompt, since that would otherwise impact all targets, I would thing

I'm inclined to think that might be more of an issue with the wrapper generated by rules_js, or even with bazel in handling string quoting in the transition from windows arguments constructed by bazel into windows batch scripts, and then from batch into the call to bash than necessarily an issue here

https://github.com/bazelbuild/bazel/issues/17487 might be a related issue
